.github/workflows/branch-push.yml

```diff
@@ -44,4 +45,81 @@ jobs:
         git add */VARIABLES.md
         git commit -s -m "Add VARIABLES.md files"
         git push
-      working-directory: ${{ github.workspace }}
\ No newline at end of file
+      working-directory: ${{ github.workspace }}
+
+  parse-turtle:
+    name: Check syntax of all changed Turtle files
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Check for changes in .ttl files
+        id: check_ttl
+        run: |
+          git fetch origin main
+          CHANGED_TTL=$(git diff --name-only origin/main -- '*.ttl')
+          if [[ -n "$CHANGED_TTL" ]]; then
+            echo "TTL_HAS_CHANGED=true" >> $GITHUB_ENV
+            echo "CHANGED_TTL=$CHANGED_TTL" >> $GITHUB_ENV
+          else
+            echo "TTL_HAS_CHANGED=false" >> $GITHUB_ENV
+            echo "Subsequent steps will be skipped as no .ttl files have changed."
+          fi
+
+      - name: Set up Python
+        if: env.TTL_HAS_CHANGED == 'true'
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12.0
+
+      - name: Install rdflib
+        if: env.TTL_HAS_CHANGED == 'true'
+        run: pip install rdflib

```

Comment:
Can we add the requirements to a file in order to install conviniently locally?
```
    - name: Install dependencies
      run: pip install -r requirements_develop.txt
```

Reply:
added requirements_ci.txt and linked it in the ci